src/main/java/com/spotify/dns/ChangeNotifiers.java

```diff
@@ -28,10 +30,38 @@
 
 public final class ChangeNotifiers {
 
+  /**
+   * Ensure that we get a unique empty set that you can't create any other way
+   * (i.e. ImmutableSet.of() always returns the same instance).
+   *
+   * This is needed to distinguishing the initial state of change notifiers from
+   * when they have gotten proper data.
+   */
+  private static final Set INITIAL_EMPTY_DATA = Collections.unmodifiableSet(new HashSet());
+
   private ChangeNotifiers() {
   }
 
   /**
+   * Use this to determine if the data you get back from a notifier is the initial result of the result of a proper
+   * DNS lookup. This is useful for distinguishing a proper but empty DNS result from the case
+   * where a lookup has not completed yet.
+   * @param set
+   * @return true if the input is an initially empty set.
+   */
+  public static <T> boolean isInitialEmptyData(Set<T> set) {
```

Comment:
Yes, this is needed for consumers to determine if they should ignore the data or not